common/dynamicconfig/file_based_client.go

```diff
@@ -187,7 +189,7 @@ func (fc *fileBasedClient) GetDurationValue(
 }
 
 func (fc *fileBasedClient) UpdateValue(name Key, value interface{}) error {
-	keyName := Keys[name]
+	keyName := name.String()
 	currentValues := make(map[string][]*constrainedValue)
 
 	confContent, err := ioutil.ReadFile(fc.config.Filepath)
```

Comment:
at a glance this seems like it's missing the same-type check as well, like the configstore client